### ISSUE TITLE
do_After click cancel is not good

### ISSUE THREAD

**Issue by DogZeroX** [space-wizards/space-station-14#15391]
## Description

Sometimes out of hurry you click twice, or lag delays your action so you click multiple times to make sure it's through, there's multiple reasons you might want to click more than once to make sure you're doing the thing you want, that why do_after click interrupt is annoying, in the end you end up fiddling by clicking twice on your and clicking more out of confusion, it's worst during stressful situations.
**Reproduction**

**Screenshots**

**Additional context**

**Comment by deltanedas** [comment]
yeah and, in general, if you need to cancel a doafter its easier to move than reclick whatever

**Comment by ElectroJr** [comment]
I personally much prefer the default behaviour being able to just cancel accidental delayed interactions. FWIW, it is trivial to customize this for each DoAfter and there probably are some that should be uncancellable.

>  lag delays your action so you click multiple times to make sure it's through

The fix for this is to make sure all the interactions get moved to shared code, which has to be done anyway. That way the user receives instant feedback.

**Comment by SlamBamActionman** [comment]
I am going to deny this issue because I think the issue is less that double-clicking cancels doafters, and more that there is poor feedback that the second click will cancel the doafter. 

I observed a streamer yesterday try to put a hat on someone and not only did he cancel the action repeatedly, he was also clicking around on other slots just trying to get something to work. I don't think the double-click was the issue there, just that the UI itself had no feedback. 

Maybe some icon next to the cursor to indicate when an action starts a doafter? Worth workshopping
